utils_test.go

```diff
@@ -42,3 +50,202 @@ func TestValidStoragePathFormat(t *testing.T) {
 		assert.NilError(t, err)
 	}
 }
+
+func TestRootlessRuntimeDir(t *testing.T) {
```

Comment:
From a quick skim, I don’t see this hitting the “`tmpDir` does not exist, trying to create it, and racing against an attacker” case. That would require even more parametrization of the implementation under test (and indirectly making the test more different from the actual system behavior, so I’m not sure how much we need that. What do others think?

Reply:
> racing against an attacker

If you will describe in in a strict way that it will be implemented.

Reply:
Consider how
```go
if err := os.Rename(newTmpDir, tmpDir); err != nil {
```
can fail if `tmpDir` has been created in the meantime.

But, again, I’m not at all sure adding a test for this one is worth it. It is certainly _doable_ but the cost (adding an extra pair of channels for communicating with the test?) may be making the code too difficult to read; the security-critical bit is ultimately the `isRootlessRuntimeDirOwner` check after the `Rename`.

Reply:
Yes, that's sounds reasonable. We need to use `tmpDir` only if renamed successfully. It will be fixed.